### DISCUSSION TITLE
process.env is empty

### DISCUSSION THREAD

**Discussion by aakoch** [parcel-bundler/parcel#6788] [Q&A] [answered]
I'm was trying to use process.env for removing some log statements when built for production, but the property is never being populated. With `watch` or `build` it's the same. The process.env property is there, just always empty.

I created a branch to test with Yarn and got the same thing. 

I'm using Pug so maybe that is messing with it? I have:
src/index.pug:
```
- var title = 'Software Developer and Clean Code Advocate'
block variables
doctype html
html.no-js(lang="en")

  head
    include head
  body
    #landing-header
      include img/header_name.svg
    
    #landing-app

    include footer

    block body_scripts

    script(src="js/landing.js" type="module")
```

And then in landing.js:
```
import { createApp } from "vue";
import App from "../components/landing";
import 'bootstrap'
import '@fortawesome/fontawesome-free/css/all.css'
import process from 'process';
console.dir(process)

const landing = createApp(App)
landing.mount('#landing-app')
```

And the console looks like:
```
{
addListener: ƒ Ji()
argv: []
binding: ƒ (e)
browser: true
chdir: ƒ (e)
cwd: ƒ ()
emit: ƒ Ji()
env: {}
...
}
```

Is this an edge case or something? Anything I should try?

Thanks,
Adam

**Comment by mischnic** [comment, marked as answer]
https://v2.parceljs.org/features/node-emulation/#%F0%9F%8C%B3-environment-variables:

> Accessing the `process.env` object directly is not supported, but accessing specific variables on it like `process.env.API_KEY` will provide the expected value.

**Reply by aakoch** [reply]
Thanks! There was one other thing I missed: 
Use the built-in process Node.js global, i.e. don't do import process from "process". 

With those 2 changes, things are working.

Thanks Niklas! 🎉
